### ISSUE TITLE
Driver-only ECC: BN wrap-up

### ISSUE THREAD

**Issue by mpg** [Mbed-TLS/mbedtls#7757]
For the release:

- [x] add ChangeLog entries
- [x] update `docs/driver-only-builds.md`
- [x] In `ssl.h`, things that currently depend on `MBEDTLS_PK_HAVE_ECC_KEYS` should probably depend on `ECP_C` instead - those are deprecated, so we don't need to extend their scope and can keep them only when `ECP_C`, as they were in the previous release.
- [x] Fix description or p256-m entry points in `docs/psa-driver-example-and-guide.md`.
- [x] Make `P256M_DRIVER_ENABLED` auto-enable the corresponding `ACCEL` macros.

Misc clean-ups that are too small to deserve separate issues, but could wait after the release:

- [x] ~Remove now-redundant `all.sh` components based on `psa_crypto_config_accel_all_curves_except_one`~ done by #8075
- [x] Remove now-redundant tests based on `syms.sh` in other `all.sh` components.
- [x] In `pk_internal`, I think the new functions `mbedtls_pk_ec_ro` and `mbedtls_pk_ec_rw` should depend on `!MBEDTLS_PK_USE_PSA_EC_DATA` as they don't really make sense with `MBEDTLS_PK_USE_PSA_EC_DATA`. → deferred to https://github.com/Mbed-TLS/mbedtls/issues/8263

**Depends on:** #7452, #7756
